src/gleam_stdlib.mjs

```diff
@@ -370,24 +372,28 @@ export function utf_codepoint_to_int(utf_codepoint) {
   return utf_codepoint.value;
 }
 
-export function regex_check(regex, string) {
-  return regex.test(string);
+export function regex_check(regex_constructor, string) {
+  return regex_constructor().test(string);
 }
 
 export function compile_regex(pattern, options) {
   try {
     let flags = "gu";
     if (options.case_insensitive) flags += "i";
     if (options.multi_line) flags += "m";
-    return new Ok(new RegExp(pattern, flags));
+    const regex_constructor = function () {
+      return new RegExp(pattern, flags);
+    };
+    regex_constructor();
```

Comment:
Not the best perf but making `/g` optional makes regex.scan explode on regular gleam regexp "objects".

For a faster non-immutable/stateful variant people can write gleam/js specific community libs?

Reply:
I think this is too much of a performance hit. What other ways are there to avoid this problem? Can we reset the mutable state?

Reply:
@lpil turns out we can <3